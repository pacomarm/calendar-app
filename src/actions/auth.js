import Swal from "sweetalert2";
import { fetchConToken, fetchSinToken } from "../helpers/fetch"
import { types } from "../types/types";


export const startLogin = (email, password) => {
    return async(dispatch) => {
        const res = await fetchSinToken('auth', {email, password}, 'POST');
        const body = await res.json();
        
        if( body.ok ){
            localStorage.setItem( 'token', body.token );
            localStorage.setItem( 'token-init-date', new Date().getTime() );
            dispatch( login({
                uid: body.uid,
                name: body.name
            }) )
        } else {
            Swal.fire('Error', 'Incorrect user credentials!', 'error')
        }
    }
}

export const startRegister = (name, email, password) => {
    return async(dispatch) => {
        const res = await fetchSinToken('auth/new', {name, email, password}, 'POST');
        const body = await res.json();

        if (body.ok){
            localStorage.setItem( 'token', body.token );
            localStorage.setItem( 'token-init-date', new Date().getTime() );
            dispatch( login({
                uid: body.uid,
                name: body.name
            }))
        } else {
            Swal.fire('Error', body.msg || 'Could not register user!', 'error')
        }
    }
}

export const startChecking = () => {
    return async(dispatch) => {

        try{
            const res = await fetchConToken('auth/renew');
            const body = await res.json();
    
            if (body.ok){
                localStorage.setItem( 'token', body.token );
                localStorage.setItem( 'token-init-date', new Date().getTime() );
                dispatch( login({
                    uid: body.uid,
                    name: body.name
                }));
    
            } else {
                dispatch( checkingFinish() )
                throw new Error(body.msg || 'Error!');
            }
        } catch(e){
            console.log(e.msg || 'Error al validar token');
        }

    }
}

const checkingFinish = () => ({ type: types.authCheckingFinish })

const login = (user) => ({
    type: types.authLogin,
    payload: user
})

export const startLogout = () => {
    return async(dispatch) => {
        localStorage.clear();
        dispatch( logout() )
    }
}

const logout = () => ({ type: types.authLogout })