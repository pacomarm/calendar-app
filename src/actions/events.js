import { fetchConToken } from "../helpers/fetch"
import { prepareEvents } from "../helpers/prepareEvents";
import { types } from "../types/types"

export const eventStartAddNew = (event) => {
    return  async(dispatch, getState) => {
        const {uid, name} = getState().auth;
        try{
            const res = await fetchConToken( 'events', event, 'POST' );
            const body = await res.json();

            if( body.ok ){
                event.id = body.event.id;
                event.user = { _id: uid, name:  name}
                dispatch( eventAddNew(event) )
            }

        } catch (e){
            console.log(e);
        }
    }
}

const eventAddNew = (event) => ({
    type: types.eventAddNew,
    payload: event
})

export const eventSetActive = (event) => ({
    type: types.eventSetActive,
    payload: event
})

export const eventClearActiveEvent = () => ({ type: types.eventClearActiveEvent })

export const eventUpdated = (event) => ({
    type: types.eventUpdate,
    payload: event
})

export const eventDeleted = () => ({ type: types.eventDeleted })

export const eventStartLoading = () => {
    return async(dispatch) => {
        try{
            const res = await fetchConToken( 'events', 'GET' );
            const body = await res.json();
            console.log(prepareEvents(body.events));
            const events = prepareEvents(body.events);

            dispatch( eventLoaded(events) )

        } catch (e){
            console.log(e);
        }
    }
}

const eventLoaded = (events) => ({
    type: types.eventLoaded,
    payload: events
})
