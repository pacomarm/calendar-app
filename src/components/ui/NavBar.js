import React from 'react'

export const NavBar = () => {
    return (
        <div className="navbar navbar-dark bg-dark mb-4">
            <span className="navbar-brand">Paco</span>
            <button className="btn btn-outline-danger">
                Salir
                <i className=" ml-2 fas fa-sign-out-alt"></i>
            </button>
        </div>
    )
}