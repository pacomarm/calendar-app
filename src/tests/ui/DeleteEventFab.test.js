import { mount, configure } from 'enzyme'
import React from 'react'
import { Provider } from 'react-redux'

import configureStoreÂ from 'redux-mock-store'
import thunk from 'redux-thunk'
import '@testing-library/jest-dom'
import { DeleteEventFab } from '../../components/ui/DeleteEventFab'

import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { eventStartDelete } from '../../actions/events'
configure({adapter: new Adapter()});

jest.mock('../../actions/events', () => ({
    eventStartDelete: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configureStore(middlewares)

const initState = {};
let store = mockStore( initState);

Storage.prototype.setItem = jest.fn();
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab/>
    </Provider>
)

describe('tests for DeleteEventFab', () => {
    
    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot(); 
    })

    test('should call eventStartDelete when clicked', () => {
        wrapper.find('button').prop('onClick')();
        expect(eventStartDelete).toHaveBeenCalled()
    })
})
