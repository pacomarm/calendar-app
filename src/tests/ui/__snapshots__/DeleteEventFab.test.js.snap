// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for DeleteEventFab should match snapshot 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className=" fas fa-trash"
  />
</button>
`;
